### ISSUE TITLE
[Feature Request] Implement test for overfitting in experimental version of optimizer

### ISSUE THREAD

**Issue by user1823** [open-spaced-repetition/fsrs4anki#259]
> > Regarding overfitting - why not split data into 2 datasets, "training" and "test" set? Optimize parameters on training set, then see how big the loss is on the test set. If test set loss is much higher than training set loss, that means the model is overfitted. That's a standard practice in ML, in fact, I'm surprised you haven't done it yet.
> 
> Because it will cost more time of optimization. In the start of this project, I fixed the structure of model, so it didn't need to test.

Now that we are continuously changing the structure of the model, I think that the risk of overfitting is large. So, it would be better if we implement the test for overfitting. What do you think?

_Originally posted by @user1823 in https://github.com/open-spaced-repetition/fsrs4anki/issues/239#issuecomment-1546021426_

You might be thinking that it is much of a chore to implement this test in every branch. But, you don't need to implement it in every branch. You can just implement the test in one separate branch and then we can change the formulas in that branch to test those models.

**Comment by Expertium** [comment]
I also think this should be implemented.

**Comment by L-M-Sherlock** [comment]
This issue has been solved by cross-validation in the branch: https://github.com/open-spaced-repetition/fsrs4anki/tree/Expt/new-baseline
